npm_and_yarn/lib/dependabot/npm_and_yarn/file_fetcher.rb

```diff
@@ -22,28 +22,32 @@ class FileFetcher < Dependabot::FileFetchers::Base # rubocop:disable Metrics/Cla
 
       # Npm always prefixes file paths in the lockfile "version" with "file:"
       # even when a naked path is used (e.g. "../dep")
-      NPM_PATH_DEPENDENCY_STARTS = %w(file:).freeze
+      NPM_PATH_DEPENDENCY_STARTS = T.let(%w(file:).freeze, [String])
       # "link:" is only supported by Yarn but is interchangeable with "file:"
       # when it specifies a path. Only include Yarn "link:"'s that start with a
       # path and ignore symlinked package names that have been registered with
       # "yarn link", e.g. "link:react"
-      PATH_DEPENDENCY_STARTS = %w(file: link:. link:/ link:~/ / ./ ../ ~/).freeze
+      PATH_DEPENDENCY_STARTS = T.let(%w(file: link:. link:/ link:~/ / ./ ../ ~/).freeze,
+                                     [String, String, String, String, String, String, String, String])
```

Comment:
8-tuple, not an array, because below we do **, which only works with a statically-known length.